doc/tls.md

```diff
@@ -0,0 +1,88 @@
+# KUKSA.val TLS concept
+
+This page describes the TLS support in KUKSA.val
+
+## Security concept
+
+KUKSA.val supports TLS for connection between KUKSA.val Databroker/Server and clients.
+
+General design concept in short:
+
+* KUKSA.val Server and KUKSA.val Databroker by default only accept TLS connection. Insecure connections can be allowed by a configuration setting
+* Mutual authentication not supported, i.e. KUKSA.val Server and KUKSA.val Databroker does not authenticate clients
+* A set of example certificates and keys exist in the [kuksa_certificates](kuksa_certificates) repository
+* The example certificates are used as default by some applications
+* The example certificates shall only be used during development and re not suitable for production use
+* KUKSA.val does not put any additional requirements on what certificates that are accepted, default settings as defined by OpenSSL and gRPC are typically used
+
+## Example certificates
+
+For more information see the [README.md](kuksa_certificates/README.md).
+
+**NOTE: The example keys and certificates shall not be used in your production environment!  **
+
+## Examples using example certificates
+
+This section intends to give guidelines on how you can verify TLS functionality with KUKSA.val.
+It is based on using the example certificates.
+
+
+## KUKSA.val databroker
+
+KUKSA.val Databroker supports TLS, but not mutual authentication.
+By default (for historical reasons) an insecure connection is used.
+In the future this might be change so that `--insecure` is required to start an insecure connection.
+
+To use a secure connection specify `--tls-cert`and `--tls-private-key`
+
+```
+~/kuksa.val/kuksa_databroker$ cargo run --bin databroker -- --metadata ../data/vss-core/vss_release_4.0.json --tls-cert ../kuksa_certificates/Server.pem --tls-private-key ../kuksa_certificates/Server.key
```

Comment:
I'd say it is the preferred one because it's easier for testing. And we mention the docker things in our quickstart. Butmentioning both wouldn't hurt.

Reply:
I sort of agree that "cargo run" is more a development kind of view (also important), but there should be an answer for people "just downloaded/pulled a release", which also would act as guidance how e.g. projects like Velocitas would do it, when they provision everthing as containers

Reply:
I think this is part of a bigger discussion on what different deployment alternatives we support out of the box. In the future we might for instance release Databroker as a Debian/Apertis package as well and then we might need to have yet another alternative documented. I added a Docker example below.